platforms/jvm/ear/src/main/java/org/gradle/plugins/ear/descriptor/internal/DefaultDeploymentDescriptor.java

```diff
@@ -343,6 +276,26 @@ public void execute(XmlProvider xmlProvider) {
         return this;
     }
 
+    private DefaultEarWebModule newDefaultEarWebModule(String path, String contextRoot) {
+        DefaultEarWebModule module = objectFactory.newInstance(DefaultEarWebModule.class);
+        module.getPath().set(path);
+        module.getContextRoot().set(contextRoot);
+        return module;
+    }
+
+    private DefaultEarModule newDefaultEarModule(String path) {
+        DefaultEarModule module = objectFactory.newInstance(DefaultEarModule.class);
+        module.getPath().set(path);
+        return module;
+    }
+
+    private DefaultEarSecurityRole newDefaultEarSecurityRole(String roleName, String description) {
+        DefaultEarSecurityRole role = objectFactory.newInstance(DefaultEarSecurityRole.class);
+        role.getRoleName().set(roleName);
+        role.getDescription().set(description);
+        return role;
+    }
+
```

Comment:
❓ This looks fishy. Shouldn't this be handled by a `DomainObjectContainer` or something instead?

Reply:
This are basically just a helper methods to set properties. I don't think `DomainObjectContainer` handles this, since we don't inject values to `constructor` but we set properties via `.set()`

Reply:
Is it an option though to inject via constructor parameters, and use a `DOC`? I am always wary about naked use of `ObjectFactory` or any other direct instantiation. These cases should be handled via regualr injection.